packages/api/src/routes/medical/document.ts

```diff
@@ -24,34 +30,31 @@ const router = Router();
  * @param req.query.patientId Patient ID for which to list documents.
  * @param req.query.dateFrom Optional start date that docs will be filtered by (inclusive).
  * @param req.query.dateTo Optional end date that docs will be filtered by (inclusive).
+ * @param req.query.organization Optional name of the contained Organization to filter docs
+ *    by (partial match and case insentitive).
+ * @param req.query.practitioner Optional name of the contained Practitioner to filter docs
+ *    by (partial match and case insentitive).
  * @return The available documents, including query status and progress - as applicable.
  */
 router.get(
   "/",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
     const patientId = getFromQueryOrFail("patientId", req);
-    const dateFrom = parseISODate(getFrom("query").optional("dateFrom", req));
-    const dateTo = parseISODate(getFrom("query").optional("dateTo", req));
+    const { dateFrom, dateTo, organization, practitioner } = getDocSchema.parse(req.query);
 
     // Confirm the CX can access this patient
     await getPatientOrFail({ cxId, id: patientId });
 
-    const documents = await getDocuments({ cxId, patientId });
-    const documentsDTO = toDTO(documents).flatMap(doc => {
-      if (!doc.indexed) return doc;
-      if (
-        (!dateFrom || dayjs(dateFrom).isBefore(doc.indexed)) &&
-        (!dateTo || dayjs(dateTo).isAfter(doc.indexed))
-      ) {
-        return doc;
-      }
-      return [];
+    const documents = await getDocuments({
+      cxId,
+      patientId,
+      dateRange: { from: dateFrom ?? undefined, to: dateTo ?? undefined },
+      organizationName: organization ? organization : undefined,
```

Comment:
nit: could simplify to organization ?? undefined. Same for practitioner in the next line